tests/playwright-tests/smoke.test.ts

```diff
@@ -0,0 +1,130 @@
+import { Page, expect, test } from '@playwright/test'
+
+const mockResponse = async (page: Page, data: Record<string, any>) => {
+  await page.route(
+    'http://fake-url.com/graphql',
+    (route) => {
+      // delay 200ms
+      setTimeout(() => {
+        route.fulfill({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify({
+            data: data,
+          }),
+        })
+      }, 50)
+    },
+    { times: 1 }
+  )
+}
+
+const mockErrorResponse = async (page: Page) => {
+  await page.route(
+    'http://fake-url.com/graphql',
+    (route) => {
+      // delay 200ms
+      setTimeout(() => {
+        route.fulfill({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify({
+            error: new Error('error'),
+          }),
+        })
+      }, 50)
+    },
+    { times: 1 }
+  )
+}
+
+test('smoke', async ({ page }) => {
+  await mockResponse(page, { count: 0 })
+  await page.goto('/smoke')
+  // Suspended on initial loading
+  await expect(page.getByText('loading')).toBeVisible()
+  await expect(page.getByText('count: 0')).toBeVisible()
+
+  await mockResponse(page, { count: 1 })
+  // Reexecutes but with wrong policy (cache-first) hence no update
+  await page.getByText('refetch-wrong-policy').click()
+  // Data shouldn't change
+  await expect(page.getByText('count: 0')).toBeVisible()
+  // Reexecute with network-only policy, hence update
+  await page.getByText('refetch-network-only').click()
+  // Updated
+  // TODO: This is another problem. For some reason stale state on network-only
```

Comment:
This is the case I'm talking about of refetching stale status not being reported.